react/I18n/format.jsx

```diff
@@ -3,26 +3,31 @@ import { DEFAULT_LANG } from '.'
 import formatDistanceToNow from 'date-fns/formatDistanceToNow'
 
 const locales = {}
-let lang = DEFAULT_LANG
+let lang = DEFAULT_LANG === 'en' ? 'en-US' : DEFAULT_LANG
 
 const getWarningMessage = lang =>
   `The "${lang}" locale isn't supported by date-fns. or has not been included in the build. Check if you have configured a ContextReplacementPlugin that is too restrictive.`
 
 export const provideDateFnsLocale = (userLang, defaultLang = DEFAULT_LANG) => {
-  lang = userLang
+  const resolvedDefaultLang = defaultLang === 'en' ? 'en-US' : defaultLang
+  const resolvedUserLang = userLang === 'en' ? 'en-US' : userLang
   try {
-    locales[defaultLang] = require(`date-fns/locale/${defaultLang}/index.js`)
+    locales[
+      resolvedDefaultLang
+    ] = require(`date-fns/locale/${resolvedDefaultLang}/index.js`)
   } catch (err) {
-    console.warn(getWarningMessage(defaultLang))
+    console.warn(getWarningMessage(resolvedDefaultLang))
   }
-
-  if (lang && lang !== defaultLang) {
+  if (resolvedUserLang && resolvedUserLang !== resolvedDefaultLang) {
     try {
-      locales[lang] = require(`date-fns/locale/${lang}/index.js`)
+      locales[
+        resolvedUserLang
+      ] = require(`date-fns/locale/${resolvedUserLang}/index.js`)
     } catch (e) {
-      console.warn(getWarningMessage(lang))
+      console.warn(getWarningMessage(resolvedUserLang))
     }
   }
+  lang = resolvedUserLang
   return locales[lang]
 }
 
```

Comment:
It also means that the consuming app should change the call to `<I18n` to pass the right ISO code. Today this call is based on what the stack / flagship are injecting. For instance https://github.com/cozy/mespapiers/blob/48b5ec048b00d29c418f06606cc9ec15c2a203c7/src/targets/browser/setupApp.jsx#L15